### PR TITLE
clowns are clumsy even when antagonist

### PR DESCRIPTION
**Pull request by Intigracy**
Not all thread posts are shitposts.

### CODE REVIEW

**Comment by ComicIronic** [comment]
Why

**Comment by Imagebard** [comment]
Because Intigracy fucks bitches
(That is, mostly himself)

**Comment by Intigracy** [comment]
Don't talk about your mother that way.

https://boards.4chan.org/vg/thread/160827380#p160875280 for Comic's question.

**Comment by ComicIronic** [comment]
This unfairly gimps clown antags, though.

**Comment by Intigracy** [comment]
For the just part only when it comes to firearms.

**Comment by ComicIronic** [comment]
And using handcuffs/legcuffs, flashing people (which is a pretty important gamemode thing), attacking people (since a lot of items have hit-yourself clumsy code, including eswords), using grenades, and more besides. You even have chances to weaken yourself in unarmed combat.

I have to ask: did you even do a minor search for what M_CLUMSY affected?

**Comment by clusterfack** [comment]
Are you sure this even fixes it? Because I'm pretty sure I put something to make antagonists not have genetic defects somewhere a long time ago.

**Comment by Intigracy** [comment]
I feel that playing clown shouldn't be made easier just because you're an antagonist.

Hilarity often happens with clowns who're nonantags like when they get fights and have the gun explode on them while trying to shoot the nuke op.

It's not gimping antag clowns it's just keeping business as usual.

**Comment by Intigracy** [comment]
Also I'll give this more work next time I get free time, since as you can see the clumsy check var isn't where I thought it was.

**Comment by Shadowmech88** [comment]
>unfairly gimps clown antags

Isn't it the player's own choice to have both clown and antag enabled?
If they don't want to be a clumsy antag, then they could disable antag while playing clown, or vice versa.

**Comment by ComicIronic** [comment]
It's not their teammates choice in a group of antags to have a clown antag who brings the team down by having a mutation.

**Comment by ghost** [comment]
>Let's make the clown not a clown when he has to murder people!

Christ, at that point it's just a skin. Vox still have to deal with N2, Mimes still have to deal with being silent, plasmamen still have to depend on plasma and their suit staying on. Keep the clown clumsy. If they wanna be murderclowns, they have to be a fucking clown.

**Comment by kol1th** [comment]
>super hard mode for clown vampire/changeling for no apparent reason

No thank you

**Comment by JustSumBody** [comment]
>Mimes still have to deal with being silent

But they have a button that lets them not be silent that they can activate of their own free will any time.

A better compromise would just be letting people choose whether or not they get clumsy when they antag clown(so you valid clowns based on whether or not they're actually being shitters rather than whether or not they have clumsy), but then you'd probably want to restrict it to solo antags to prevent fucking over team antags like Comic mentioned.

This is also still a code solution to a player/administration problem which is disgusting.

**Comment by SonixApache** [comment]
Clowns should be funny.

File: code/game/objects/items/weapons/melee/energy.dm [hunk 1]
```diff
@@ -51,6 +51,7 @@
 	flags = FPRINT
 	origin_tech = Tc_MAGNETS + "=3;" + Tc_SYNDICATE + "=4"
 	attack_verb = list("attacks", "slashes", "stabs", "slices", "tears", "rips", "dices", "cuts")
+	clumsy_check = 1
```
Comment (ComicIronic) [inline code/game/objects/items/weapons/melee/energy.dm]: Needs to be `var/clumsy_check`, and please use `TRUE` and `FALSE`.

Comment (Intigracy) [reply code/game/objects/items/weapons/melee/energy.dm]: var/clumsy_check exists at some level on weapons, I'll probably make a seperate PR to set it as a weapon variable and to turn all the clumsy checks into a proc as opposed to copypaste.

**Comment by Intigracy** [comment]
@clusterfack where would I find your change?

**Comment by SangriaMonkey** [comment]
There is literally nothing wrong with aggressive gun-wielding clowns.
Clumsy is a worse meme than airflow or high fantasy beast races.

**Comment by Sprok0** [comment]
Reminder that clowns can remove their clumsy by just getting a rye pill from medbay

**Comment by TheFriggenSawbones** [comment]
@Imagebard you're fuckin perma'd why are you here

**Review by PJB3005 — APPROVED** [review]

**Comment by IratePirate** [comment]
Why the hell was this ever changed? The Clown is supposed to be ludicrous even when antag, not a ultra tacticool slaughterer with a revolver. The revolver is supposed to have a chance of blowing up in his face for maximum hilarity.

**Comment by ComicIronic** [comment]
This barely counts as "changed", it was added to TGcode back in the SVN days: https://github.com/d3athrow/vgstation13/commit/d5388d6d04cb8093d55483e8961c3f87106a3377

And why should the Clown be fucked over as an antag? You can't just say that's how it's supposed to be.

**Comment by SonixApache** [comment]
@ComicIronic because that's how it's supposed to be

**Comment by IratePirate** [comment]
That's exactly how it's supposed to be. The Clown was even originally a punishment role.

If a Clown is playing the round as a hyper-competent then they're the un-funny cancer killing the serb, no matter whether they're antag or not.

**Comment by Shadowmech88** [comment]
Not that I'm arguing for or against the PR with this, but original intent should not always govern current policy. The tastes and opinions of a playerbase can change over time, and changes to a feature should not be restricted on the sole basis of "this was the intent when the feature was added years ago, therefore it should not be changed now."

Whether a thing *was* supposed to be a certain way when it was added does not necessarily mean that said thing should be that way *now*.

**Comment by Intigracy** [comment]
This is clearly exactly like US law.

**Comment by Allspice** [comment]
As thing stands the clown role is for assholes that want to get away with more grif than they "should" so long as it involves peels or lube.

I don't think this is ideal but oh well.

**Comment by ComicIronic** [comment]
The textual reading supports the idea that you are all idiots.

**Comment by SangriaMonkey** [comment]
Thanks.

**Comment by Probe1** [comment]
Make a server pool

**Comment by Probe1** [comment]
We are completely divided on this issue and it requires being sent to the players to decide.

**Comment by Probe1** [comment]
http://ss13.moe/index.php/poll/78

**Comment by Shadowmech88** [comment]
@Probe1 
>This poll addresses the following PR https://github.com/d3athrow/vgst

The poll doesn't seem to have been made correctly.

**Comment by Probe1** [comment]
Yes we can out of characters and the input box doesn't have a catch to let us know.

**Review by Probe1 — DISMISSED** [review]
Write an exception for ``` /obj/item/weapon/melee/energy/sword/bsword ```

It's literally a clown only esword.

**Comment by ComicIronic** [comment]
Server vote has been made-u. Probe, is your request going to block, or can you make a PR for it later?

**Comment by Probe1** [comment]
yeah it's just gonna cause headaches down the line if we don't.

**Comment by Intigracy** [comment]
See above.

**Comment by Intigracy** [comment]
https://github.com/d3athrow/vgstation13/blob/bacedc18184a3c0b40c0119888148f2e35eacc35/code/game/objects/items/weapons/melee/energy.dm#L162

**Comment by carshalash** [comment]
Coders who dont play the game refuse to test things and bugs ensure. Why am I not fucking surprised.

**Comment by Probe1** [comment]
@Intigracy banana sword still activates clumsy

**Comment by Shadowmech88** [comment]
When you turn the sword on, or when you attempt to attack with it?

**Comment by carshalash** [comment]
Attack with it.

Wait, you fucks do realize that clowns have clumsy effect melee attacks too right?

**Comment by TheFriggenSawbones** [comment]
the clown not having clumsy while an antag is less of a flavor thing and more allowing them to function as antags.

This is a really dumb change and I doubt the playerbase understood how much it fucked over clown players.

**Comment by TheFriggenSawbones** [comment]
@Intigracy I'd ask you to test your code but that would mean you'd have to play the game.

**Comment by ghost** [comment]
@Intigracy 

antag modes this ruins for multiple antags: Revolutionary, Cultist.  You now have a team member who can't harm others consistently, FLASH others, or cuff others.

antag modes this ruins for the clown: Vampire, Traitor, Changeling.  They have a specific item they can not use now, wow.  Also, a changeling/vampire that can not cuff (unsure of this part, APPARENTLY clumsy code is actually broken at the moment and clowns can cuff anyone regardless of it not being intended) or consistently harm others for these specific modes may as well just ghost at round start.

This is so incredibly short-sighted, poorly thought out, and untested.

**Comment by SonixApache** [comment]
Or clowns could just do exactly as their name implies and clown around. Stealth antags may not be the exciting high octane gameplay you expect from ss13, but it's definitely possible.

Treat clowns as clowns, always.

**Comment by PJB3005** [comment]
~No, it was tested.~

~The server vote was hugely in favour of making clowns clumsy. We listen to the server vote.~

~Had we not we'd have people thread shitposting constantly.~

Nevermind me I thought people were complaining that the change in general was dumb, didn't see that the banana esword was fucked.

**Comment by PJB3005** [comment]
http://ss13.moe/index.php/poll/78

**Comment by Sprok0** [comment]
Clowns absolutely do not deserve nice things, if they wanted to be capable they wouldn't have picked clown.

**Comment by TheFriggenSawbones** [comment]
I feel as if this pr only serves to make the game less fun for certain players, and it also screws over team antag gamemodes if you're unlucky enough to get a clown as a starting rev/cult.

TG-mans reasoning seems to be extremely flawed and the 4chan thread he cited is 404'd.

jesus

**Comment by Intigracy** [comment]
https://dl.dropboxusercontent.com/s/f2d7752iu61j4fk/2017-01-01_12-35-23.mp4

**Comment by Intigracy** [comment]
Handcuffs not working is not even close to being my fault, I didn't touch that code in this PR.

**Comment by Sprok0** [comment]
It was specifically the banana bunch, not the normal banana sword

**Comment by ghost** [comment]
lame, should atleast give the clown clumsy-proof weapons on their uplinks

**Comment by Shadowmech88** [comment]
The banannon is clumsy-proof.

**Pull request merged**